### DISCUSSION TITLE
Can argo fail parallel steps if one fails

### DISCUSSION THREAD

**Discussion by GAHila** [argoproj/argo-workflows#8044] [Q&A]
We have a workflow that has over a hundred parallel steps, loaded withParam dynamically, parallelism is 15 but we want that if one fails argo won't continue to the next group or steps to run. They are technically not dependent on each other and are executed in parallel for speed but we'd want argo to call the onExit hook rather than pick up more steps. Any way that can be achieved at all or do we have to modify the design for this to happen?

**Comment by terrytangyuan** [comment]
Check out the discussions and example in this issue that might address your use case: https://github.com/argoproj/argo-workflows/issues/3644.

**Reply by GAHila** [reply]
Looks like it does what I'm looking but it broke onExit. I'd assumed onExit would execute either way. Will look around more.
